core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -590,6 +600,11 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
         |                              the node running the Application Master via the Secure
         |                              Distributed Cache, for renewing the login tickets and the
         |                              delegation tokens periodically.
+        |
+        | Kubernetes only:
+        |  --kubernetes-namespace NS   The namespace in the Kubernetes cluster within which the
+        |                              application must be launched. The namespace must already
+        |                              exist in the cluster. (Default: default).
```

Comment:
We should also add check for `validateKillArguments` and `validateStatusRequstArguments`.

Reply:
Done.

Reply:
now that docker image is without a default value (`spark.kubernetes.*.docker.image`), (yes that discussion is ongoing) I wonder if it makes sense to bubble that up as a `--param` for visibility/convenient.

tbh I'm generally against adding `--param` to submit because of the potential confusion it can cause, but since we are here and there's a `--kubernetes-namespace`

Reply:
I think if we eventually decide to not have default docker images, we should make the options  `--param` ones. I'm not sure if we want to make a call and do that in this PR though. Can we defer this to a later time when we are clearer on how we publish and maintain the images?